core/server/master/src/main/java/alluxio/master/meta/MetaDailyBackup.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.meta;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.underfs.UfsStatus;
+import alluxio.underfs.UnderFileSystem;
+import alluxio.underfs.UnderFileSystemConfiguration;
+import alluxio.util.FormatUtils;
+import alluxio.util.ThreadFactoryUtils;
+import alluxio.util.URIUtils;
+import alluxio.util.io.PathUtils;
+import alluxio.wire.BackupOptions;
+import alluxio.wire.BackupResponse;
+
+import com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.temporal.ChronoUnit;
+import java.util.TreeMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Backing up primary master metadata everyday at a fixed UTC time.
+ */
+public final class MetaDailyBackup {
+  private static final Logger LOG = LoggerFactory.getLogger(MetaDailyBackup.class);
+
+  private final Pattern mBackupPattern
```

Comment:
can we put this constant close to the place where we choose backup names? I'm worried that if we change the format, this might break

Reply:
@aaudiber I put the following two constants in `MetaDailyBackup`:
```
  public static final String BACKUP_FILE_FORMAT = "alluxio-backup-%s-%s.gz";
  private static final Pattern BACKUP_FILE_PATTERN
      = Pattern.compile("alluxio-backup-[0-9]+-[0-9]+-[0-9]+-([0-9]+).gz");
```

I first think of putting them in the `DefaultMetaMaster`, but I want to keep `DefaultMetaMaster` constants as less as possible. Should I put them in the `BackupManager`?